internal/pkg/term/selector/local_selector.go

```diff
@@ -128,23 +129,80 @@ func (s *staticSelector) askCron(scheduleValidator prompt.ValidatorFunc) (string
 // localFileSelector selects from a local file system where a workspace does not necessarily exist.
 type localFileSelector struct {
 	prompt        Prompter
+	ws            *workspace.Workspace
 	fs            *afero.Afero
 	workingDirAbs string
 }
 
 // NewLocalFileSelector constructs a localFileSelector.
-func NewLocalFileSelector(prompt Prompter, fs afero.Fs) (*localFileSelector, error) {
+func NewLocalFileSelector(prompt Prompter, fs afero.Fs, ws *workspace.Workspace) (*localFileSelector, error) {
 	workingDirAbs, err := os.Getwd()
 	if err != nil {
 		return nil, fmt.Errorf("get working directory: %w", err)
 	}
 	return &localFileSelector{
 		prompt:        prompt,
+		ws:            ws,
 		fs:            &afero.Afero{Fs: fs},
 		workingDirAbs: workingDirAbs,
 	}, nil
 }
 
+// StaticSources asks the user to select from a list of directories and files in the current directory and two levels down.
+func (s *localFileSelector) StaticSources(selPrompt, selHelp, anotherPathPrompt, anotherPathHelp string, pathValidator prompt.ValidatorFunc) ([]string, error) {
+	dirsAndFiles, err := s.listDirsAndFiles()
+	if err != nil {
+		return nil, err
+	}
+	if len(dirsAndFiles) == 0 {
+		log.Warningln("No directories or files were found in your workspace. Enter a relative path with the 'custom path' option if you'd like to use a hidden file.")
+	}
+	dirsAndFiles = append(dirsAndFiles, []string{staticSourceUseCustomPrompt}...)
+	var selections []string
+	selections, err = s.prompt.MultiSelect(
+		selPrompt,
+		selHelp,
+		dirsAndFiles,
+		nil,
+		prompt.WithFinalMessage(staticAssetsFinalMsg),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("select directories and/or files: %w", err)
+	}
+	for i, selection := range selections {
+		if selection == staticSourceUseCustomPrompt {
+			for {
+				customPath, err := s.prompt.Get(
+					anotherPathPrompt,
+					anotherPathHelp,
+					pathValidator,
+					prompt.WithFinalMessage(customPathFinalMsg))
+				if err != nil {
+					return nil, fmt.Errorf("get custom directory or file path: %w", err)
+				}
+				if selection == staticSourceUseCustomPrompt {
```

Comment:
isn't this _always_ true at this point? based on the condition above?

Reply:
If the user inputs a custom path, `selection` now == the custom path. So if they opt to enter another and the for-loop repeats, the path will be added to the list differently than the first custom path.

But this all proves WX's point that the code is hard to read, and I'm changing it to her suggestion!:)